pkg/hub/submarineragent/controller.go

```diff
@@ -371,23 +377,37 @@ func (c *submarinerAgentController) syncSubmarinerConfig(ctx context.Context,
 
 	// prepare submariner cluster environment
 	errs := []error{}
-	cloudProvider, preparedErr := cloud.GetCloudProvider(c.kubeClient, c.manifestWorkClient, c.eventRecorder, managedClusterInfo, config)
-	if preparedErr == nil {
-		preparedErr = cloudProvider.PrepareSubmarinerClusterEnv()
-	}
+	var condition metav1.Condition
+	if config.Status.ManagedClusterInfo.Platform != "GCP" {
+		cloudProvider, preparedErr := cloud.GetCloudProvider(c.restConfig, c.kubeClient, c.manifestWorkClient, c.dynamicClient,
+			nil, c.eventRecorder, managedClusterInfo, config)
+		if preparedErr == nil {
+			preparedErr = cloudProvider.PrepareSubmarinerClusterEnv()
+		}
 
-	condition := metav1.Condition{
-		Type:    configv1alpha1.SubmarinerConfigConditionEnvPrepared,
-		Status:  metav1.ConditionTrue,
-		Reason:  "SubmarinerClusterEnvPrepared",
-		Message: "Submariner cluster environment was prepared",
-	}
+		condition = metav1.Condition{
+			Type:    configv1alpha1.SubmarinerConfigConditionEnvPrepared,
+			Status:  metav1.ConditionTrue,
+			Reason:  "SubmarinerClusterEnvPrepared",
+			Message: "Submariner cluster environment was prepared",
+		}
 
-	if preparedErr != nil {
-		condition.Status = metav1.ConditionFalse
-		condition.Reason = "SubmarinerClusterEnvPreparationFailed"
-		condition.Message = fmt.Sprintf("Failed to prepare submariner cluster environment: %v", preparedErr)
-		errs = append(errs, preparedErr)
+		if preparedErr != nil {
+			condition.Status = metav1.ConditionFalse
+			condition.Reason = "SubmarinerClusterEnvPreparationFailed"
+			condition.Message = fmt.Sprintf("Failed to prepare submariner cluster environment: %v", preparedErr)
+			errs = append(errs, preparedErr)
+		}
+	} else {
+		if !helpers.IsSubmarinerEnvPrepared(c.configClient, config.Namespace, config.Name) {
+			condition = metav1.Condition{
+				Type:   configv1alpha1.SubmarinerConfigConditionEnvPrepared,
+				Status: metav1.ConditionFalse,
+				Reason: "SubmarinerClusterEnvPreparationNotYetDone",
+			}
+		} else {
+			return operatorhelpers.NewMultiLineAggregate(errs)
+		}
```

Comment:
I don't think we should or need to update the condition here to indicate it's pending.  The spoke is responsible for  updating the status in this case. Also we could get a race where the pending status overrides the completed status.   

```suggestion
		return nil
```

Reply:
Now using nil value for conditon.